backend/src/products/product.controller.ts

```diff
@@ -0,0 +1,84 @@
+import { Request, Response } from 'express'
+import productModel from './product.model'
+class productController {
+  getAllProducts = async (req: Request, res: Response) => {
+    const products = await productModel.find({})
+    return res.json(products)
+  }
+
+  getProductById = async (req: Request, res: Response) => {
+    const product = await productModel.findById(req.params.id)
+    if (product) return res.json(product)
+    res.status(404).json({ message: 'product not found' })
+  }
+  deleteProduct = async (req: Request, res: Response) => {
+    try {
+      const id = req.params.id
+      const product = await productModel.findByIdAndDelete(id)
+      if (product) {
+        return res.status(200).json({ message: 'Product removed' })
+      }
+      res.status(400)
+      throw new Error('product not found')
+    } catch (error) {
+      res.status(500)
+      throw new Error(error.message)
+    }
+  }
+
+  createProduct = async (req: Request, res: Response) => {
+    const {
+      name,
+      brand,
+      category,
+      price,
+      countInStock,
+      image,
+      description
+    }: {
+      name: string
+      brand: string
+      category: string
+      price: number
+      countInStock: number
+      image: string
+      description: string
+    } = req.body
+
+    try {
+      const product = new productModel({
+        name,
+        brand,
+        category,
+        price,
+        countInStock,
+        image,
+        description,
+        user: req.user._id
+      })
+      const newProduct = await product.save()
+      res.status(200).json(newProduct)
+    } catch (error) {
+      res.status(401).send({ message: error.message })
```

Comment:
this status code 401 should be 400 because it's a valdatio error
you could read more about that from https://developer.mozilla.org/en-US/docs/Web/HTTP/Status